frontend/src/concepts/notificationPoller/NotificationPollerContext.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+import { AppNotification } from '~/redux/types';
+import useNotification from '~/utilities/useNotification';
+
+export type NotificationPollerContextType = {
+  watchForNotification: (item: NotificationPollerItem) => Promise<void>;
+};
+
+type NotificationPollerContextProviderProps = {
+  children: React.ReactNode;
+};
+
+export type FinalNotificationPollerResponse =
+  | ({ status: 'success' | 'error' } & Pick<AppNotification, 'title' | 'message' | 'actions'>)
+  | { status: 'stop' };
+
+export type NotificationPollerResponse = FinalNotificationPollerResponse | { status: 'repoll' };
+
+export type NotificationCallback = (signal: AbortSignal) => Promise<NotificationPollerResponse>;
+
+export type NotificationPollerItem = {
+  delayRepollMs?: number;
+  callback: NotificationCallback;
+};
+
+export const NotificationPollerContext = React.createContext<NotificationPollerContextType>({
+  watchForNotification: () => Promise.resolve(),
+});
+
+const DEFAULT_REPOLL_INTERVAL_MS = 0;
+
+export const NotificationPollerContextProvider: React.FC<
+  NotificationPollerContextProviderProps
+> = ({ children }) => {
+  const notification = useNotification();
+  const abortControllersMapRef = React.useRef(new Map<NotificationPollerItem, AbortController>());
+  const timeoutIdsMapRef = React.useRef(
+    new Map<NotificationPollerItem, ReturnType<typeof setTimeout>>(),
+  );
+
+  const invoke = React.useCallback(
+    async (itemToInvoke: NotificationPollerItem, signal: AbortSignal) => {
+      const delayRepollMs = itemToInvoke.delayRepollMs ?? DEFAULT_REPOLL_INTERVAL_MS;
+      const response = await itemToInvoke.callback(signal);
```

Comment:
I'm wondering if we don't just delay it out of the gate 🤔 

I can't imagine an immediate invoke will ever help -- as it's "right in front of you" per se... maybe it's an option in case we have some "Command and Control" interface some day where someone may want immediate statuses on several items?

Changing it on this line may be awkward given the other structure of this function... but I think "delaying on first invoke" has value and I imagine ideally would be the preferred state.